Scripts/Game/UI/Modded/ModifiedGroupsManagerComponent.c

```diff
@@ -0,0 +1,295 @@
+				//---------------------------------------------------------------------------------------------------------------------------------
+				
+				// Check if target player is in a vehicle.
+				CompartmentAccessComponent compartmentAccess = CompartmentAccessComponent.Cast(localplayer.FindComponent(CompartmentAccessComponent));
+				if (compartmentAccess)
+				{
+					// Check target players current compartment.
+				    BaseCompartmentSlot compartment = compartmentAccess.GetCompartment();  
+				    if (compartment)
+				    {
+						// Check target players current compartment type, then assign his icon.
+				        ECompartmentType compartmentType = SCR_CompartmentAccessComponent.GetCompartmentType(compartment);
+						switch (compartmentType)
+						{
+							// Passanger/Commander 
+							// ToDo: impliment Commander compartment detection and icon when/if it's added to the game
+							case ECompartmentType.Cargo : {
+								iconArray.Insert("{B910A93F355F168C}Layouts\UI\Textures\Icons\imagecargo_ca.edds");
+								break;
+							};
+							// Driver
+							case ECompartmentType.Pilot : {
+								iconArray.Insert("{C2B2F451FB157A89}Layouts\UI\Textures\Icons\imagedriver_ca.edds");
+								
+								break;
+							};
+							// Gunner
+							case ECompartmentType.Turret : {
+								iconArray.Insert("{3DAAB773C8C29812}Layouts\UI\Textures\Icons\imagegunner_ca.edds");
+								break;
+							};
+						};
+				    };
+				};
+				
+				//---------------------------------------------------------------------------------------------------------------------------------
+				// Override regular Icons If Set By SL
+				//---------------------------------------------------------------------------------------------------------------------------------
+				
+				string playerOverideIcon =  ReturnUIValue(localPlayerID, "Icon");
+				
+				if (playerOverideIcon && iconArray.IsEmpty()) {
+					switch (playerOverideIcon) 
+						{
+						case "Team Lead" : {
+							iconArray.Insert("{6D45BA2CCC322312}Layouts\UI\Textures\Icons\iconmanteamleader_ca.edds");
+							break;
+						};
+						case "Medic" : {
+							iconArray.Insert("{01F2523A4EE5C48B}Layouts\UI\Textures\Icons\iconmanmedic_ca.edds");
+							break;
+						};
+						case "Sniper" : {
+							iconArray.Insert("{318B797C57BE3C29}Layouts\UI\Textures\Icons\iconmansniper_ca.edds");
+							break;
+						};
+						case "Machine Gunner" : {
+							iconArray.Insert("{CCF40410BDB53870}Layouts\UI\Textures\Icons\iconmanmg_ca.edds");
+							break;
+						};		
+						case "Anti-Tank" : {
+							iconArray.Insert("{DC86195B44F5A345}Layouts\UI\Textures\Icons\iconmanat_ca.edds");
+							break;
+						};
+						case "Grenadier" : {
+							iconArray.Insert("{B7757F2024A3DC87}Layouts\UI\Textures\Icons\iconmangrenadier_ca.edds");
+							break;
+						};
+						case "Man" : {
+							iconArray.Insert("{71ED761DF5BA041C}Layouts\UI\Textures\Icons\iconman_ca.edds");
+							break;
+						};
```

Comment:
Can we turn these into variables? Will be easier for maintenance

Reply:
Yeah, will do

Reply:
Updated